Automattic/wp-calypso
# Add storage addon to cart in /start/business signup flow

Related to pau2Xa-4MT-p2.

## Proposed Changes

* This PR adds the ability to include a storage addon as a signup dependency, so that it can be applied at checkout.
* The storage addon dependency is enabled for the `/start/business` flow.
* For some context, we are introducing storage addons in the landing page pricing grids in D128661-code. This PR will enable a user to carry over their selection on the landing page pricing grid to the signup flow. Users can select either a 50GB or 100GB storage addon, on the Business plan or Commerce plan on the pricing grid.
* Since the ecommerce plan on the pricing grid takes to the Stepper flow, a separate PR will follow to implement these changes to Stepper's tailored ecommerce flow.

**Technical implementation notes**

I decided to go with the approach of adding a ["silent" step](https://github.com/Automattic/wp-calypso/pull/85968/files#diff-2b3d29b1d06e979107ec149ebb7dc18fb9c4f789b74759550f12fa4ea84d55c7) to the `/start/business` flow. This step does not render anything, it's sole purpose is to check whether a storage addon has been supplied to the query param (through its fulfilled step callback), and if the addon exists then it is added to the cart. This gives the flexibility to add this step to any other flow that could expect a storage addon in the query parameter. 

## Testing Instructions



* Visit `/start/business/?storage=50gb-storage-add-on` and go through the signup flow. Verify that the 50GB addon is automatically added at checkout. Repeat this in both logged-out and logged-in scenarios.
* Repeat previous step with the URL `/start/business/?storage=100gb-storage-add-on`.
* Visit `/start/business` and `/start/business-monthly`, and in both scenarios verify that the behaviour is unchanged. Checkout page shouldn't add any storage addons.
